swarm_listen.go

```diff
@@ -40,7 +40,12 @@ func (s *Swarm) Listen(addrs ...ma.Multiaddr) error {
 func (s *Swarm) AddListenAddr(a ma.Multiaddr) error {
 	tpt := s.TransportForListening(a)
 	if tpt == nil {
-		return ErrNoTransport
+		select {
+		case <-s.proc.Closing():
+			return ErrSwarmClosed
```

Comment:
I'm not sure I understand why this here? Which edge case is this addressing? Either we accept or we don't accept adding listen addresses while the swarm is closing/closed, more generally, but this looks weird?

Reply:
I'll add a comment: TL;DR: we could fail to get the transport either because we're closed, or because the transport was never registered.